src/psyclone/tests/psyir/transformations/extract_test.py

```diff
@@ -404,18 +337,35 @@ def test_single_node_dynamo0p3():
 
     schedule, _ = etrans.apply(schedule.children[0])
     code = str(psy.gen)
-    output = (
-        "      ! ExtractStart\n"
-        "      ! CALL write_extract_arguments(argument_list)\n"
-        "      !\n"
-        "      DO cell=1,f1_proxy%vspace%get_ncell()\n"
-        "        !\n"
-        "        CALL testkern_code(nlayers, a, f1_proxy%data, f2_proxy%data, "
-        "m1_proxy%data, m2_proxy%data, ndf_w1, undf_w1, map_w1(:,cell), "
-        "ndf_w2, undf_w2, map_w2(:,cell), ndf_w3, undf_w3, map_w3(:,cell))\n"
-        "      END DO\n"
-        "      !\n"
-        "      ! ExtractEnd\n")
+    output = """      ! ExtractStart
```

Comment:
This test should really x-fail with a link to an appropriate issue (because, e.g. not all necessary data is written out - no. of dofs, dofmaps etc.). (Although it's exciting that you've got this far!)

Reply:
Yes and no :) Not failing has the advantage that I will be alerted on any changes that affects those tests. For example, I had a few tests x-failing, which now failed because the driver name has changed, not for the original reason. With xfail this will go unnoticed.

So I have rewritten all those tests to use pytest.xfail(...), with a final assert to make sure it triggers if no test should xfail, so we will notice if an xfailing test suddenly passes. We discussed this in #593 - but while we rejected this at that time, it is very useful while I am still working on it, I need to know if the behaviour of any existing tests changes (i.e. it fails for a different reason). So this way now all tests are xfailing, but specifically only at the location that they should be failing at, the rest of tests is certain to work (or else the test will fail). Once this work is finished, the pytest.xfail() calls can be removed obviously.